drivers/pinctrl/CMakeLists.txt

```diff
@@ -11,6 +11,7 @@ zephyr_library_sources_ifdef(CONFIG_PINCTRL_ITE_IT8XXX2 pinctrl_ite_it8xxx2.c)
 zephyr_library_sources_ifdef(CONFIG_PINCTRL_NPCX pinctrl_npcx.c)
 zephyr_library_sources_ifdef(CONFIG_PINCTRL_NUMICRO pinctrl_numicro.c)
 zephyr_library_sources_ifdef(CONFIG_PINCTRL_NRF pinctrl_nrf.c)
+zephyr_library_sources_ifdef(CONFIG_SOC_R8A779F0 pfc_r8a779f0.c)
```

Comment:
why `CONFIG_SOC_...` down here?

Reply:
Because all the code inside the file is related to the specified SoC, but the pfc rcar driver can be used for other RCar SoCs as well. Do you mean that it would be better to enclose all code inside the file with `#ifdef CONFIG_SOC_R8A779F0`?

As far as I can see, some other drivers use a similar approach, such as in the following examples:

```
bluetooth/hci/CMakeLists.txt:28:zephyr_library_sources_ifdef(CONFIG_SOC_NRF5340_CPUAPP nrf53_support.c)

flash/CMakeLists.txt:69:  if(CONFIG_SOC_SERIES_STM32H7X)
flash/CMakeLists.txt:71:  elseif(CONFIG_SOC_SERIES_STM32WBAX)

clock_control/CMakeLists.txt:37:if(CONFIG_SOC_SERIES_STM32MP1X)
clock_control/CMakeLists.txt:45:elseif(CONFIG_SOC_SERIES_STM32WBAX)
clock_control/CMakeLists.txt:49:  zephyr_library_sources_ifdef(CONFIG_SOC_SERIES_STM32C0X clock_stm32c0.c)
clock_control/CMakeLists.txt:50:  zephyr_library_sources_ifdef(CONFIG_SOC_SERIES_STM32F0X clock_stm32f0_f3.c)
clock_control/CMakeLists.txt:62:  zephyr_library_sources_ifdef(CONFIG_SOC_SERIES_STM32WBX clock_stm32l4_l5_wb_wl.c)

clock_control/CMakeLists.txt:63:  zephyr_library_sources_ifdef(CONFIG_SOC_SERIES_STM32WLX clock_stm32l4_
```